src/map/ai/controllers/trust_controller.cpp

```diff
@@ -404,7 +406,61 @@ bool CTrustController::Cast(uint16 targid, SpellID spellid)
         targid = POwner->targid;
     }
 
-    return CMobController::Cast(targid, spellid);
+    auto  PTarget      = (CBattleEntity*)POwner->GetEntity(targid, TYPE_MOB | TYPE_PC | TYPE_PET | TYPE_TRUST);
+    auto  PSpellFamily = PSpell->getSpellFamily();
+    bool  canCast      = true;
+
+    static_cast<CCharEntity*>(POwner->PMaster)->ForPartyWithTrusts([&](CBattleEntity* PMember)
+    {
+        if (PMember->objtype == TYPE_TRUST && PMember->PAI->IsCurrentState<CMagicState>())
+        {
+            auto MState = static_cast<CMagicState*>(PMember->PAI->GetCurrentState());
+
+            if (MState)
+            {
+                auto MSpell       = MState->GetSpell();
+                auto MTarget      = MState->GetTarget();
+                auto MSpellFamily = MSpell->getSpellFamily();
+                auto MSpellID     = MSpell->getID();
+
+                if (PSpell->isBuff())
+                {
+                    if (PSpellFamily == MSpellFamily && spellid <= MSpellID)
+                    {
+                        canCast = false;
+                    }
+                }
+                if (PSpell->isCure())
+                {
+                    if (PTarget == MTarget && PTarget->GetHPP() > 50)
+                    {
+                        canCast = false;
+                    }
+                }
+                if (PSpell->isDebuff())
+                {
+                    if (PSpellFamily == MSpellFamily && spellid <= MSpellID)
+                    {
+                        canCast = false;
+                    }
+                }
+                if (PSpell->isNa())
+                {
+                    if (PSpellFamily == MSpellFamily && spellid == MSpellID)
+                    {
+                        canCast = false;
+                    }
+                }
+            }
+        }
+    });
+
+    if (!canCast)
+    {
+        return false;
+    }
+
+    return CController::Cast(targid, spellid);
```

Comment:
Not familiar with this code, is there a reason for the change from `CMobController` to `CController` on the return?

Reply:
To be honest, that was there before, id have to compare and see if i changed it or not, but i know there was an issue with using a one type of controller

Reply:
looks like it was left over from when it was Topaz code, ill have to change that and test it

Reply:
Ok tested with CMobController, all works good, changes pushed